packages/jaspr_content/lib/components/file_tree.dart

```diff
@@ -0,0 +1,376 @@
+            'vertical-align': 'middle',
+          },
+        ),
+        css('&.directory', [
+          css('> details', [
+            css('&').styles(
+              padding: Padding.only(left: 1.5.rem),
+              border: Border.none,
+              backgroundColor: Colors.transparent,
+            ),
+            css('> summary', [
+              css('&').styles(
+                maxWidth: 100.percent,
+                padding: Padding.symmetric(horizontal: .625.rem),
+                margin: Margin.only(left: (-1.5).rem, bottom: Unit.zero),
+              ),
+              css('&::marker').styles(
+                color: Color.variable(FileTreeTheme._iconVariable),
+              ),
+              css('&:hover', [
+                css('&').styles(
+                  cursor: Cursor.pointer,
+                  color: Color.variable(FileTreeTheme._highlightVariable),
+                ),
+                css('svg').styles(color: Color.variable(FileTreeTheme._highlightVariable)),
+                css('.highlight svg').styles(color: Color.variable(FileTreeTheme._backgroundVariable)),
+                css('~ul').styles(
+                  border: Border.only(left: BorderSide(color: Color.variable(FileTreeTheme._highlightVariable))),
+                ),
+              ]),
+            ]),
+          ]),
+        ]),
+      ]),
+      css('.tree-entry', [
+        css('&').styles(
+          display: Display.inlineFlex,
+          alignItems: AlignItems.start,
+          gap: Gap.all(0.5.rem),
+          raw: {'vertical-align': 'middle'},
+        ),
+        css('.tree-entry-name').styles(
+          display: Display.inlineFlex,
+          alignItems: AlignItems.center,
+        ),
+        css('.comment').styles(
+          padding: Padding.only(left: 1.5.rem),
+          color: Color.variable(FileTreeTheme._iconVariable),
+          fontStyle: FontStyle.italic,
+        ),
+        css('.highlight', [
+          css('&').styles(
+            padding: Padding.symmetric(horizontal: 0.25.rem),
+            radius: BorderRadius.circular(0.25.rem),
+            alignItems: AlignItems.center,
+            color: Color.variable(FileTreeTheme._backgroundVariable),
+            backgroundColor: Color.variable(FileTreeTheme._highlightVariable),
+          ),
+          css('svg').styles(
+            color: Color.variable(FileTreeTheme._backgroundVariable),
+          ),
+        ]),
+      ]),
+    ]),
+  ];
+}
+
+class _FileTree extends StatelessComponent {
+  const _FileTree({required this.items, super.key});
+
+  final List<FileTreeItem> items;
+
+  @override
+  Component build(BuildContext context) {
+    return div(classes: 'file-tree not-content', [
+      ul([
+        for (final item in items) buildFileTreeItem(item),
+      ]),
+    ]);
+  }
+
+  Component buildFileTreeItem(FileTreeItem item) {
+    if (item.children.isNotEmpty) {
+      return li(classes: 'directory', [
+        details(open: item.isOpen, [
+          summary([
```

Comment:
Good idea using `details` and `summary` here so the relationship is accessible without custom ARIA handling.